src/math/random.js

```diff
@@ -131,13 +131,12 @@ p5.prototype.random = function (min, max) {
   } else {
     rand = Math.random();
   }
-
-  if (arguments.length === 0) {
+  if (typeof min === 'undefined') {
     return rand;
   } else
-  if (arguments.length === 1) {
-    if (arguments[0] instanceof Array) {
-      return arguments[0][Math.floor(rand * arguments[0].length)];
+  if (typeof max === 'undefined') {
```

Comment:
@lmccart What're your thoughts, should we be doing anything preventative against `null` here? The old code did not do anything particular around null inputs, they would still have triggered the `arguments.length === 1` path.

That said, there's a difference between `arguments.length === 1` and `typeof secondArg === 'undefined'`; technically the former would be represented by `typeof firstArg !== 'undefined' && typeof secondArg === 'undefined'`, to capture the case where e.g. 

```js
var obj = {
  max: 7
};
random( obj.min, obj.max ); // obj.min -> undefined
```

I lean towards leaving things as they are for efficiency, since it's much more likely that `min` will be something non-mathematical that throws an error than that it will be a valid reference which is undefined, but I wanted to vet the importance of guarding against edge cases vs maintaining the existing code's efficiency.